L1Trigger/L1THGCal/test/HGCalTriggerBestChoiceTester.cc

```diff
@@ -223,88 +281,248 @@ void HGCalTriggerBestChoiceTester::rerunBestChoiceFragments(const edm::Event& e,
     } //end loop on modules
-
+    
 }
 
 
 /*****************************************************************/
-void HGCalTriggerBestChoiceTester::fillModule( const std::vector<HGCDataFrame<HGCalDetId,HGCSample>>& dataframes,
-        const HGCalBestChoiceDataPayload& fe_payload,
-        const vector<pair<HGCalDetId, uint32_t > >& linearized_dataframes)
+void HGCalTriggerBestChoiceTester::fillModule( const std::vector<HGCDataFrame<HGCalDetId,HGCSample>>& dataframes,  const vector<pair<HGCalDetId, uint32_t > >& linearized_dataframes, const HGCalBestChoiceDataPayload& fe_payload_TCsums_woBestChoice, const HGCalBestChoiceDataPayload& fe_payload_TCsums_BestChoice, const HGCalBestChoiceDataPayload& fe_payload, map <HGCalDetId,double>& simhit_energies,  map<uint32_t, double>& TC_simhit_energies)
+
 /*****************************************************************/
 {
     // HGC cells part
     size_t nHGCDigi = 0;
     unsigned hgcCellModuleSum = 0;
+    // digis, cell based info
     for(const auto& frame : dataframes)
-    {
+      {
         uint32_t value = frame[2].data();
-        if(value>0)
-        {
-            nHGCDigi++;
-            hgcCellModuleSum += value;
-            hgcCellData_->Fill(value);
+        nHGCDigi++;
+        hgcCellModuleSum += value;
+        hgcCellData_->Fill(value);
+        if (is_Simhit_comp_){
+          double sim_energy= simhit_energies[(HGCalDetId)frame.id()];
+          if (sim_energy >0){
+            hgcCellData_SimHitasso_->Fill(value);
+            hgcCellSimHits_->Fill(sim_energy);
+            hgcCellData_vsSimHits_->Fill(sim_energy,value);
+          }
         }
-    }
+      }
     hgcCellsPerModule_->Fill(nHGCDigi);
     hgcCellModuleSum_->Fill(hgcCellModuleSum);
-
+    
+    // linearized samples, cell based info
     for(const auto& frame : linearized_dataframes){
-        hgcCellData_linampl_-> Fill(frame.second);
+      hgcCellData_linampl_-> Fill(frame.second);
+      if (is_Simhit_comp_){
+        double sim_energy= simhit_energies[(HGCalDetId)frame.first];
+        if (sim_energy >0){ 
+          hgcCellData_linampl_vsSimHits_-> Fill(sim_energy,frame.second);
+          hgcCellData_linampl_vsSimHits_zoom_-> Fill(sim_energy,frame.second);
+        }
+      }
     }
-
+    
     // trigger cells part
+    // after sum, no best choice, no encode/decode
+    int ncells_noBestChoice = 0;
+    for(const auto& tc : fe_payload_TCsums_woBestChoice.payload)
+    {
+      if(tc>0)
+        {
+          triggerCellData_noBestChoice_->Fill(tc);
+          if (is_Simhit_comp_){
+            if (TC_simhit_energies[ncells_noBestChoice] >0){
+              triggerCellSimHits_noBestChoice_->Fill(TC_simhit_energies[ncells_noBestChoice]);
+              triggerCellData_noBestChoice_vsSimHits_->Fill(TC_simhit_energies[ncells_noBestChoice],tc);
+            }
+          }
+        }
+      ncells_noBestChoice++;
+    }
+    
+    // after sum, best choice, no encode/decode
+    int ncells_BestChoice = 0;
```

Comment:
The name 'ncells...' is misleading, the variable is not used as a number of cells but as a map index